sources/representation/qt/mainform.cpp

```diff
@@ -179,19 +184,47 @@ void MainForm::startGameLoop(int id, QString map)
     QElapsedTimer process_performance;
     qint64 max_process_time = 0;
 
+    int delta_time = 0;
+    float time_per_frame_ms = 1000 / fps_cap_;
+    float sleep_time = 0;
+    QElapsedTimer delta;
+    delta.start();
     while (true)
     {
-        ++fps_counter;
-
         process_performance.start();
-        GetRepresentation().Process();
-        GetScreen().Swap();
+        if(fps_cap_)
+        {
+            GetRepresentation().Process();
+            GetScreen().Swap();
+            ++fps_counter;
```

Comment:
Here just needed:
```
process_performance.start();
GetRepresentation().Process();
GetScreen().Swap();
++fps_counter;
qint64 process_time = process_performance.nsecsElapsed();
if (process_time > max_process_time)
{
    max_process_time = process_time;
}
int sleep_time = time_per_frame_ms - process_time / 1e6;
if (sleep_time > 0)
{
    QThread::msleep(sleep_time);
}
```
Much simpler and works if `time_per_frame_ms` less or equal than zero.